programs/mp-sol-restaking/src/instructions/admin/configure_main_vault.rs

```diff
@@ -1,52 +1,104 @@
 use crate::{error::ErrorCode, state::MainVaultState, MAX_PERFORMANCE_FEE_BP, MAX_WITHDRAW_FEE_BP};
 use anchor_lang::prelude::*;
+use anchor_spl::token::{Mint, TokenAccount};
 
-#[derive(AnchorSerialize, AnchorDeserialize, Debug)]
-pub struct ConfigureMainVaultValues {
-    unstake_ticket_waiting_hours: Option<u16>,
-    withdraw_fee_bp: Option<u16>,
-    performance_fee_bp: Option<u16>,
-    treasury_mpsol_account: Option<Pubkey>,
-    new_admin_pubkey: Option<Pubkey>,
+#[derive(Accounts)]
+pub struct ConfigureTreasuryAccount<'info> {
+    #[account()]
+    pub admin: Signer<'info>,
+
+    #[account(mut, has_one = admin, has_one = mpsol_mint)]
+    pub main_state: Account<'info, MainVaultState>,
+
```

Comment:
You don't need to input mpsol_mint as an account. You can use #[account(token::mint = main_state.mpsol_mint)] check instead